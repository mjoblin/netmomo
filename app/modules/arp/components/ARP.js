import moment from 'moment';
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Table } from 'antd';

import { getARP } from '../selectors';
import './style.scss';


export const ARP = class extends React.Component {
    render() {
        const columns = [
            {
                title: 'Operation',
                dataIndex: 'operation',
            },
            {
                title: 'Source HW',
                dataIndex: 'src_hw',
            },
            {
                title: 'Source IP',
                dataIndex: 'src_ip',
            },
            {
                title: 'Dest HW',
                dataIndex: 'dst_hw',
            },
            {
                title: 'Dest IP',
                dataIndex: 'dst_ip',
            },
            {
                title: 'Notes',
                dataIndex: 'notes',
            },
            {
                title: 'When',
                dataIndex: 'time',
                sorter: (a, b) => a.time - b.time,
                render: (text, record) =>
                    moment.unix(record.time).format('MMMM DD YYYY, h:mm:ss a'),
            },
        ];

        return (
            <div className="arp">
                <h2>ARP</h2>

                <Table
                    className="results-table"
                    dataSource={this.props.arp}
                    columns={columns}
                    pagination={false}
                    size="middle"
                />
            </div>
        );
    }
};


ARP.propTypes = {
    arp: PropTypes.array.isRequired,
};

const mapStateToProps = state => ({
    arp: getARP(state),
});


export default connect(
    mapStateToProps
)(ARP);
