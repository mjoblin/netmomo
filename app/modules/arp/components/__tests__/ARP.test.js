import React from 'react';
import { shallow } from 'enzyme';
import { ARP } from '../ARP';

const testData = [
    {
        key: 'foo',
        data: {
            dst_hw: "01:01:01:01:01:01",
            dst_ip: "192.168.1.1",
            notes: "source device is new",
            operation: "reply",
            src_hw: "02:02:02:02:02:02",
            src_ip: "192.168.1.2",
            time: 1508700000.111111,
        },
    }
];

describe('ARP', () => {
    it('matches snapshot', () => {
        const component = shallow(
            <ARP
                arp={testData}
            />
        );

        expect(component).toMatchSnapshot();
    });
});