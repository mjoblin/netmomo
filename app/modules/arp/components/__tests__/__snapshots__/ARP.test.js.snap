// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ARP matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ARP
    arp={
      Array [
        Object {
          "data": Object {
            "dst_hw": "01:01:01:01:01:01",
            "dst_ip": "192.168.1.1",
            "notes": "source device is new",
            "operation": "reply",
            "src_hw": "02:02:02:02:02:02",
            "src_ip": "192.168.1.2",
            "time": 1508700000.111111,
          },
          "key": "foo",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          ARP
        </h2>,
        <Table
          bordered={false}
          className="results-table"
          columns={
            Array [
              Object {
                "dataIndex": "operation",
                "title": "Operation",
              },
              Object {
                "dataIndex": "src_hw",
                "title": "Source HW",
              },
              Object {
                "dataIndex": "src_ip",
                "title": "Source IP",
              },
              Object {
                "dataIndex": "dst_hw",
                "title": "Dest HW",
              },
              Object {
                "dataIndex": "dst_ip",
                "title": "Dest IP",
              },
              Object {
                "dataIndex": "notes",
                "title": "Notes",
              },
              Object {
                "dataIndex": "time",
                "render": [Function],
                "sorter": [Function],
                "title": "When",
              },
            ]
          }
          dataSource={
            Array [
              Object {
                "data": Object {
                  "dst_hw": "01:01:01:01:01:01",
                  "dst_ip": "192.168.1.1",
                  "notes": "source device is new",
                  "operation": "reply",
                  "src_hw": "02:02:02:02:02:02",
                  "src_ip": "192.168.1.2",
                  "time": 1508700000.111111,
                },
                "key": "foo",
              },
            ]
          }
          indentSize={20}
          loading={false}
          locale={Object {}}
          pagination={false}
          prefixCls="ant-table"
          rowKey="key"
          rowSelection={null}
          showHeader={true}
          size="middle"
          useFixedHeader={false}
        />,
      ],
      "className": "arp",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "ARP",
        },
        "ref": null,
        "rendered": "ARP",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "bordered": false,
          "className": "results-table",
          "columns": Array [
            Object {
              "dataIndex": "operation",
              "title": "Operation",
            },
            Object {
              "dataIndex": "src_hw",
              "title": "Source HW",
            },
            Object {
              "dataIndex": "src_ip",
              "title": "Source IP",
            },
            Object {
              "dataIndex": "dst_hw",
              "title": "Dest HW",
            },
            Object {
              "dataIndex": "dst_ip",
              "title": "Dest IP",
            },
            Object {
              "dataIndex": "notes",
              "title": "Notes",
            },
            Object {
              "dataIndex": "time",
              "render": [Function],
              "sorter": [Function],
              "title": "When",
            },
          ],
          "dataSource": Array [
            Object {
              "data": Object {
                "dst_hw": "01:01:01:01:01:01",
                "dst_ip": "192.168.1.1",
                "notes": "source device is new",
                "operation": "reply",
                "src_hw": "02:02:02:02:02:02",
                "src_ip": "192.168.1.2",
                "time": 1508700000.111111,
              },
              "key": "foo",
            },
          ],
          "indentSize": 20,
          "loading": false,
          "locale": Object {},
          "pagination": false,
          "prefixCls": "ant-table",
          "rowKey": "key",
          "rowSelection": null,
          "showHeader": true,
          "size": "middle",
          "useFixedHeader": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            ARP
          </h2>,
          <Table
            bordered={false}
            className="results-table"
            columns={
              Array [
                Object {
                  "dataIndex": "operation",
                  "title": "Operation",
                },
                Object {
                  "dataIndex": "src_hw",
                  "title": "Source HW",
                },
                Object {
                  "dataIndex": "src_ip",
                  "title": "Source IP",
                },
                Object {
                  "dataIndex": "dst_hw",
                  "title": "Dest HW",
                },
                Object {
                  "dataIndex": "dst_ip",
                  "title": "Dest IP",
                },
                Object {
                  "dataIndex": "notes",
                  "title": "Notes",
                },
                Object {
                  "dataIndex": "time",
                  "render": [Function],
                  "sorter": [Function],
                  "title": "When",
                },
              ]
            }
            dataSource={
              Array [
                Object {
                  "data": Object {
                    "dst_hw": "01:01:01:01:01:01",
                    "dst_ip": "192.168.1.1",
                    "notes": "source device is new",
                    "operation": "reply",
                    "src_hw": "02:02:02:02:02:02",
                    "src_ip": "192.168.1.2",
                    "time": 1508700000.111111,
                  },
                  "key": "foo",
                },
              ]
            }
            indentSize={20}
            loading={false}
            locale={Object {}}
            pagination={false}
            prefixCls="ant-table"
            rowKey="key"
            rowSelection={null}
            showHeader={true}
            size="middle"
            useFixedHeader={false}
          />,
        ],
        "className": "arp",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "ARP",
          },
          "ref": null,
          "rendered": "ARP",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "bordered": false,
            "className": "results-table",
            "columns": Array [
              Object {
                "dataIndex": "operation",
                "title": "Operation",
              },
              Object {
                "dataIndex": "src_hw",
                "title": "Source HW",
              },
              Object {
                "dataIndex": "src_ip",
                "title": "Source IP",
              },
              Object {
                "dataIndex": "dst_hw",
                "title": "Dest HW",
              },
              Object {
                "dataIndex": "dst_ip",
                "title": "Dest IP",
              },
              Object {
                "dataIndex": "notes",
                "title": "Notes",
              },
              Object {
                "dataIndex": "time",
                "render": [Function],
                "sorter": [Function],
                "title": "When",
              },
            ],
            "dataSource": Array [
              Object {
                "data": Object {
                  "dst_hw": "01:01:01:01:01:01",
                  "dst_ip": "192.168.1.1",
                  "notes": "source device is new",
                  "operation": "reply",
                  "src_hw": "02:02:02:02:02:02",
                  "src_ip": "192.168.1.2",
                  "time": 1508700000.111111,
                },
                "key": "foo",
              },
            ],
            "indentSize": 20,
            "loading": false,
            "locale": Object {},
            "pagination": false,
            "prefixCls": "ant-table",
            "rowKey": "key",
            "rowSelection": null,
            "showHeader": true,
            "size": "middle",
            "useFixedHeader": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
